File: src/main/java/com/yoghurt1131/weatherapi/domain/input/valueobject/RangedWeather.java
package com.yoghurt1131.weatherapi.domain.input.valueobject;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class RangedWeather {

    // unix timestamp
    @JsonProperty("dt")
    private long time;

    @JsonProperty("dt_text")
    private String utcDatetime;

    @JsonProperty("main")
    private RangedWeatherProperty property;

    @JsonProperty("weather")
    private List<Weather> weathers;

    @JsonProperty("clouds")
    private Clouds clouds;

    @JsonProperty("wind")
    private Wind wind;

}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/domain/input/valueobject/RangedWeather.kt
package dev.yoghurt1131.weatherapi.domain.input.valueobject

import com.fasterxml.jackson.annotation.JsonProperty

data class RangedWeather(
    @JsonProperty("dt") val time: Long,
    @JsonProperty("dt_txt") val utcDatetime: String,
    @JsonProperty("main") var property: RangedWeatherProperty,
    @JsonProperty("weather") var weathers: List<Weather>,
    @JsonProperty("clouds") var clouds: Clouds,
    @JsonProperty("wind") var wind: Wind
)
